File: nodeClasses/scenario.m
classdef scenario < handle    % derive, not less or equal
    
    
    % simulationScenario = scenario(numSteps,'fixed',3);  example
  
    properties     
        numSteps;
        scenarioType;  % 'ncorn' or 'fixed'
        numIntervals;
        intervalSize;
    end
    
    methods
        
        %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
        % Constructor
        %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
        function obj = scenario(numSteps,scenarioType,numIntervals)
            
            obj.numSteps = numSteps;
            obj.numIntervals = numIntervals;
            obj.scenarioType = scenarioType;
            obj.intervalSize = numSteps/numIntervals;   % redundency 
            
            
            
        end
        
        function initializeScenario(obj,nodes,indicies)
        
            numNodes = length(indicies);    % index
            for i=1:numNodes
                nodes{indicies(i)}.txProbability = 0;
            end
       
            
        end
        
        function updateScenario(obj,nodes, indicies, steptime)
            
            numNodes = length(indicies);
            
            switch obj.scenarioType
                
                case 'ncorn'
               
                    currentInterval = floor(steptime/obj.intervalSize);
                    if steptime > obj.numSteps*(obj.numIntervals-1)/obj.numIntervals
                        currentInterval = obj.numIntervals;
                    end
                    
                    
                     for i=1:numNodes
                        nodes{indicies(i)}.txProbability = currentInterval/obj.numIntervals;        % -j ??                       
                     end
                    
                     
                     
                     
                case 'fixed'    % nothing in fixed
                    
                otherwise 
                    error('scenario.Type is unknown');
            end
            
        end
        
        

    end  
end

File: scenario.py
#!/usr/bin/env python2
# -*- coding: utf-8 -*-
"""
Created on Sun Apr  1 22:06:32 2018

@author: Jet
"""
import math

class scenario:
    numSteps = [ ]
    scenarioType = [ ]
    numIntervals = [ ]
    intervalSize = [ ]
    
    def __init__(self,numSteps,scenarioType,numIntervals):
        self.numSteps     = numSteps
        self.numIntervals = numIntervals
        self.scenarioType = scenarioType
        self.intervalSize = numSteps/numIntervals
        
    def initializeScenario(self,nodes,indicies):
        numNodes = len(indicies)   
        for i in range(0,numNodes):
            nodes[indicies(i)-1].txProbability = 0   # how to get access to node
            # notice -1 or not, to be debugged

    def updateScenario(self,nodes, indicies, steptime):
        numNodes = len(indicies)          
        if self.scenarioType == 'ncorn':
            currentInterval = math.floor(steptime/self.intervalSize)
            if steptime > self.numSteps * (self.numIntervals-1) / self.numIntervals:
                currentInterval = self.numIntervals                    
            for i in range(0,numNodes):
                nodes[indicies(i)-1].txProbability = currentInterval/self.numIntervals        # -j ??                       
        elif self.scenarioType   == 'fixed':   #% nothing in fixed
            pass # todo                    
        else: 
            print "scenario.Type is unknown"